### ISSUE TITLE
Bvb_Grid_Source_Doctrine2->getTotalRecords() =>NonUniqueResultException with groupBy

### ISSUE THREAD

**Issue by imonteiro** [zfdatagrid/grid#183]
_Original author: david.qu...@gmail.com (August 18, 2011 12:54:01)_

<b>What steps will reproduce the problem?</b>
[code]
$qb = $this-&gt;_em-&gt;createQueryBuilder()
-&gt;select('log.id')
-&gt;from('Entity\TErrorLog','log')
-&gt;groupBy('log.id');

$grid = $this-&gt;_helper-&gt;grid();
$source = new Bvb_Grid_Source_Doctrine2($qb);
$grid-&gt;setSource($source);
$this-&gt;view-&gt;grid = $grid-&gt;deploy(); =&gt; crash
[/code]
<b>What is the expected output? What do you see instead?</b>
[code]
Exception de type Doctrine\ORM\NonUniqueResultException
#0 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Doctrine\ORM\Internal\Hydration\AbstractHydrator.php(99): Doctrine\ORM\Internal\Hydration\SingleScalarHydrator-&gt;_hydrateAll()
#1 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Doctrine\ORM\AbstractQuery.php(594): Doctrine\ORM\Internal\Hydration\AbstractHydrator-&gt;hydrateAll(Object(Doctrine\DBAL\Driver\PDOStatement), Object(Doctrine\ORM\Query\ResultSetMapping), Array)
#2 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Doctrine\ORM\AbstractQuery.php(459): Doctrine\ORM\AbstractQuery-&gt;execute(Array, 4)
#3 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Doctrine\ORM\AbstractQuery.php(485): Doctrine\ORM\AbstractQuery-&gt;getSingleResult(4)
#4 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Bvb\Grid\Source\Doctrine2.php(1146): Doctrine\ORM\AbstractQuery-&gt;getSingleScalarResult()
#5 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Bvb\Grid.php(2779): Bvb_Grid_Source_Doctrine2-&gt;getTotalRecords()
#6 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Bvb\Grid\Deploy\Table.php(1592): Bvb_Grid-&gt;deploy()
#7 F:\ZEND\APPLICATIONS\exotica\zf1d2\application\modules\secret\controllers\MailingController.php(72): Bvb_Grid_Deploy_Table-&gt;deploy()
#8 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Zend\Controller\Action.php(513): Secret_MailingController-&gt;listeAction()
#9 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Zend\Controller\Dispatcher\Standard.php(295): Zend_Controller_Action-&gt;dispatch('listeAction')
#10 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Zend\Controller\Front.php(954): Zend_Controller_Dispatcher_Standard-&gt;dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#11 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Zend\Application\Bootstrap\Bootstrap.php(97): Zend_Controller_Front-&gt;dispatch()
#12 F:\ZEND\APPLICATIONS\exotica\zf1d2\library\Zend\Application.php(366): Zend_Application_Bootstrap_Bootstrap-&gt;run()
#13 F:\ZEND\APPLICATIONS\exotica\zf1d2\application\public\index.php(27): Zend_Application-&gt;run()
#14 {main}

[/code]

<b>Please insert the appropriate values;</b>
                    Zend Framework version: 1.11
ZFDatgrid Version (Bvb_Grid::getVersion()): latest
                          Operating system: WinXP
                               PHP Version: 5.3
               Database Server and version: MySql 5
                          Source Adatapter: Bvb_Grid_Source_Doctrine2

<b>Please provide any additional information below.</b>

_Original issue: http://code.google.com/p/zfdatagrid/issues/detail?id=745_

**Comment by imonteiro** [comment]
_From None on September 06, 2011 13:28:05_
This issue was closed by revision r1863.

**Comment by imonteiro** [comment]
_From nach...@gmail.com on February 02, 2012 23:40:43_
Hi Bento. I'm having another exception not being catch by this TRY/CATCH block.
Doctrine\ORM\NoResultException

Here's my solution:

try {
            $count = $qb->getQuery()->getSingleScalarResult();
        } catch (NonUniqueResultException $e) {
            //when the result is non unique its most likely that a group by was used
            //if so, we just get the complete result and count the number of results
            //fixes issue #745
            $result = $qb->getQuery()->getResult();
            $count = count($result);

```
    } catch (Doctrine\ORM\NoResultException $e) {
        $count = 0;
    }
```
